listenerutil/parse.go

```diff
@@ -93,6 +94,12 @@ type ListenerConfig struct {
 	CorsAllowedOrigins                       []string    `hcl:"cors_allowed_origins"`
 	CorsAllowedHeaders                       []string    `hcl:"-"`
 	CorsAllowedHeadersRaw                    []string    `hcl:"cors_allowed_headers"`
+
+	// Custom Http response headers
+	CustomApiResponseHeaders    map[int]map[string][]string `hcl:"-"`
```

Comment:
There is an overall issue in the file which is that header names are case-insensitive but the code here won't treat it that way. Go handles this properly if you use `http.Header`. I think reading them in from HCL as strings is probably fine but once you are mixing and matching values below (which I think actually should not be done) then probably actions should take place on an http.Header.

Reply:
so this is an interesting one. I thought the same, but it seems somewhere (internal to the hcl parser?) the header names *are* normalized. you can see this happening in the test. `cache-control` values overwrites the `Cache-Control` value, and the `test` values are normalized to `Test`

Reply:
Ah, I didn't look at the function below -- it's happening in `parseHeaders` -- the `textproto.CanonicalMIMEHeaderKey` call. This is the same call that `http.Header` uses internally (at least for now).

It might be nice for `parseHeaders` to return an `http.Header` which is then what is sent around for further use by the system. So `Custom(Ui)ResponseHeaders` becomes `map[int]http.Header`. Thoughts?

Reply:
I think this would be the best way to do it if we had access to the internal workings of ResponseWriter, but the interface we're given only accepts headers in a string/string format, so it would be a bit harder to use. with the map we can just iterate over the key/value pairs

Reply:
if we can still iterate over it then that would work fine. I just meant that the ResponseWriter interface just gives us access to a single http.Header via ResponseWriter.Header() so we need to use Header.Add or Header.Set no matter what.